Feature vehicles stats

peter-budo opened the pull request:
@karllindmark 

Vehicle stats available. Not much to say about it.
- Weapon stats JSON handling classes been in wrong package so moved
- Created UI and functional stuff for vehicles
- Fixed some of already existing resources and added new vehicle group string map
- Corrected imports, seems like they got messed up little for some reason

I know that groupVehicle class inside VehicleStatistics https://github.com/karllindmark/com.ninetwozero.battlelog/blob/846bea2e7ea7952cde6b7d0f721a9b95fbf44ebe/core/src/main/java/com/ninetwozero/bf4intel/json/stats/vehicles/VehicleStatistics.java#L22 is rather robust but it does quick job of sorting all vehicles in groups and setting up star count, kills and progression

peter-budo committed:
Package naming and location corrected

Seems like when I did weapons I did not reallocated core json packages in correct directory. Now it is fixed

peter-budo committed:
Sample of vehicles statistics json

peter-budo committed:
Parsing vehicle stats JSON

peter-budo committed:
Vehicle fragment setup

- Renamed VehicleStats in soldieroverview to OverviewVehicleStats to avoid confusion
- Setup VehicleStatsFragment
- Added VehicleStats grouping through GroupedVehicleStats (for the moment to follow battlelog vehicle stats template. Grouping is rather robust)

peter-budo committed:
Vehicle fragment and adapter work

- Created vehicle stats adapter that is reusing list item weapon layout (maybe renamed soon)
- Updated structure of GroupedVehicleStats and VehicleStats classes to improve data handling (group name, progression)
- SimpleGetRequest constructor with string deprecated

peter-budo committed:
Merge branch 'dev' into feature-vehicles-stats

peter-budo committed:
Finalized vehicle stats

- Updated FragmentFactory with VehicleStatsFragment
- Moved logic for generating list of GroupedVehicleStats from fragment to VehicleStatistics and added test to check number of produced items
- Renamed list_item_weapon layout to list_stats_item as both weapons and vehicles are using same layout
- Fixed missing vehicle images in the image map
- Provided string map for group names
- Fixed parsing of JSON to fetch actual value for service star progress

peter-budo committed:
Optimized imports

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/factories/UrlFactory.java:
```diff
@@ -26,6 +26,10 @@ public static URL buildWeaponStatsURL(final int soldierId, final int platformId)
         return createURL(String.format("warsawWeaponsPopulateStats/%d/%d/stats/", soldierId, platformId));
     }
 
+    public static URL buildVehicleStatsURL(final int soldierId, final int platformId) {
```
Should we keep soldierId as int or String? Do we get any benefits of keeping it as int? :-)

peter-budo replied:
Not decided yet for 100% so for now sticking with keeping `soldier Id` what truly is integer

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/vehicles/VehicleImageMap.java:
```diff
@@ -50,14 +50,16 @@
             put("WARSAW_ID_P_VNAME_Z9", R.drawable.vehicle_z9_haitun);
             put("WARSAW_ID_P_VNAME_ZBD09", R.drawable.vehicle_zbd09);
             put("WARSAW_ID_P_VNAME_ZFB05", R.drawable.vehicle_zfb05);
-// Not available yet            put("WARSAW_ID_P_XP0_VNAME_DPV", R.drawable.vehicle_xp0_dpv);
```
:+1:

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/vehicles/VehiclesGroupStringMap.java:
```diff
@@ -0,0 +1,32 @@
+
+public class VehiclesGroupStringMap {
+
+    private static final Map<String, Integer> map = new HashMap<String, Integer>() {
```
Splendid - I'll be using this on my end when it's merged :))

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/vehicles/VehicleStatsFragment.java:
```diff
@@ -0,0 +1,83 @@
+
+    private static final int ID_LOADER = 2200;
+    private ListView listView;
+    private VehicleStatistics vs;
```
Do we need to save `vs` and `adapter` at the moment? Could these be changed to an inline `final VehicleStatistics vehicleStatistics`/`final VehicleStatsAdapter adapter` inside the method instead? Just a thought. :)

peter-budo replied:
`vs` is forgotten artefact, however I believe `adapter` will be need it as soon we start to apply view states

karllindmark replied:
We should probably remove `vs`, but `adapter` is OK to keep imo. :+1:

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/weapons/WeaponStatsAdapter.java:
```diff
@@ -29,17 +29,17 @@ public long getItemId(int position) {
     public View getView(int position, View view, ViewGroup parent) {
         if (view == null) {
-            view = LayoutInflater.from(context).inflate(R.layout.list_item_weapon, parent, false);
+            view = LayoutInflater.from(context).inflate(R.layout.list_stats_item, parent, false);
```
Reusability :+1:

karllindmark commented on core/src/main/java/com/ninetwozero/bf4intel/json/stats/vehicles/VehicleStatistics.java:
```diff
@@ -0,0 +1,55 @@
+        return vehiclesList;
+    }
+
+    public List<GroupedVehicleStats> groupVehicles() {
```
`fetchGroupedVehicles()`?

peter-budo committed:
Changes per github comments

- Renamed groupedVehicles method to fetchGroupedVehicles
- Made VehicleStatistics vs local variable instead of class variable

karllindmark commented:
Not sure if anything breaking due to changes here, but I can't switch from weapons --> vehicles --> weapons - I get a blank screen when going back to weapons.

If we _add_ the unlocks-impl PR first, we can backport the ViewPager functionality into this one as well. It should probably make a difference (if not only enabling the swipe).

:moneybag:

karllindmark commented:
I noticed that the images are still low-res - do we want to update them now or in the future?

peter-budo commented:
Leave images for now

Commit:
Merge unlocks-impl changes from dev

Commit:
Update SoldierStatisticsActivity to use ViewPager

Commit:
Extract general functionality into BaseTabActivity

peter-budo committed:
Build config improvements

- Corrected version of Android Gradle Plugin from 0.6.3 to 0.7.1
- Configured Lint options to not fail on Lint errors, to produce XML report and name of the report
- Added lint-report.xml to .gitignore

peter-budo committed:
Fixed issues with tabs reloading

- The thing was that when you switched from vehicles back to weapons, fragment would attempts to create new loader. However in reality vehicles loader is returned as it was not destroyed. GSON then attempts to parse vehicle result that results in empty weapon adapter. Simple solution is resetLoader on each create.

peter-budo committed:
Merge branch 'feature-vehicles-stats' of github.com:ninetwozero/com.ninetwozero.battlelog into feature-vehicles-stats

peter-budo commented:
@karllindmark give it a go...

karllindmark commented:
Will be home in a bit, will check then :-)
On Dec 22, 2013 11:21 PM, "Peter Mikloško" notifications@github.com wrote:

> @karllindmark https://github.com/karllindmark give it a go...
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/karllindmark/com.ninetwozero.battlelog/pull/37#issuecomment-31097509
> .

karllindmark commented:
Nice catch @peter-budo! Anything else we need to address here, or merge+move on to reports?
